component/loki/write/types.go

```diff
@@ -70,6 +71,21 @@ func (r *EndpointOptions) Validate() error {
 	return nil
 }
 
+// QueueConfig controls how the queue logs remote write client is configured. Note that this client is only used when the
+// loki.write component has WAL support enabled.
+type QueueConfig struct {
+	Capacity     int           `river:"capacity,attr,optional"`
+	DrainTimeout time.Duration `river:"drain_timeout,attr,optional"`
+}
```

Comment:
If these parameters are something users can configure, then we should list them in the component docs.

Reply:
Handled in https://github.com/grafana/agent/pull/5434